internal/v2/s2av2.go

```diff
@@ -0,0 +1,99 @@
+package s2av2
+
+import (
+	"errors"
+	"context"
+	"net"
+	"crypto/tls"
+
+	"google.golang.org/grpc/credentials"
+	"github.com/google/s2a-go/internal/v2/tls_config_store"
+)
+
+const (
+	s2aSecurityProtocol = "s2av2"
+)
+
+type s2av2TransportCreds struct {
+	info     *credentials.ProtocolInfo
+	config   *tls.Config
+	isClient bool
+}
+
+// NewClientCreds returns a client-side transport credentials object that uses
+// the S2Av2 to establish a secure connection with a server.
+func NewClientCreds() (credentials.TransportCredentials, error) {
+	creds := &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		isClient: true,
+	}
+	return creds, nil
+}
+
+// NewServerCreds returns a server-side transport credentials object that uses
+// the S2Av2 to establish a secure connection with a client.
+func NewServerCreds() (credentials.TransportCredentials, error) {
+	creds := &s2av2TransportCreds{
+		info: &credentials.ProtocolInfo{
+			SecurityProtocol: s2aSecurityProtocol,
+		},
+		isClient: false,
+	}
+	return creds, nil
+}
+
+// ClientHandshake initiates a client-side mTLS handshake using the S2Av2.
+func (c *s2av2TransportCreds) ClientHandshake(ctx context.Context, serverAuthority string, rawConn net.Conn) (_ net.Conn, _ credentials.AuthInfo, err error) {
```

Comment:
Per offline discussion, we decided to test the success flows for `ClientHandshake` and `ServerHandshake` by locally starting up a client and server and having them talk to one another using the new transport credentials.

Reply:
Will do; I'll be sure to test in the next PR when we create an example go client / server using these s2av2transport credentials.